### PR TITLE
Maintenance work

### LINKED ISSUE #35: Github workflows
Do you want me to update the `.github` folder to use actions to build everything?

### LINKED ISSUE #36: DataFrames upper version
Is there a reason for the `^0.20` upper limit on the `DataFrames` version? I am trying to plug the results of `NCBITaxonomy` into a tree, but the set of versions don't overlap. I'm willing to attempt a fix that would let us use the most recent `DataFrames` release.

### LINKED ISSUE #37: Use Documenter recent version
The project uses an old version of Documenter - do you want me to update it to the next version (as part of #35 maybe?)?

### PR DESCRIPTION
**Pull request by tpoisot**
This is _Work in Progress_ to update some of the build system, documentation, etc. 

Closes #37 
Closes #36 
Closes #35

### CODE REVIEW

**Comment by tpoisot** [comment]
Giving up for now.

**Comment by mkborregaard** [comment]
What's the issue @tpoisot ?

**Comment by tpoisot** [comment]
> What's the issue @tpoisot ?

The tests don't run because they can't load JuliaDB, even though it's in the test/Project.toml

**Comment by mschauer** [comment]
Is this the issue?
```
205 WARNING: using OnlineStats.transform in module JuliaDB conflicts with an existing identifier.
206 * Testing LinkTree.jl: Error During Test at /home/runner/work/Phylo.jl/Phylo.jl/test/runtests.jl:19
207 Got exception outside of a @test
208 LoadError: ArgumentError: Package IterableTables not found in current path:
209  - Run `import Pkg; Pkg.add("IterableTables")` to install the IterableTables package.
```

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+jobs:
+  test:
+    name: Julia ${{ matrix.version }} - R ${{ matrix.R }} - ${{ matrix.os }} 
+    runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.experimental }}
+    strategy:
+      matrix:
+        version:
+          - '1.0'
```
Comment (mschauer) [inline .github/workflows/CI.yml:17]:
```suggestion
```

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          - '1.1'
```
Comment (mschauer) [inline .github/workflows/CI.yml:18]:
```suggestion
```

1.0 and 1.1 are not compatible with a Project.toml in the ./test directory

Comment (tpoisot) [reply .github/workflows/CI.yml:18]: well that explains that

**Comment by mschauer** [comment]
Sorry,
> The princess is in another castle

```
  LoadError: ArgumentError: Package Random not found in current path:
250
  - Run `import Pkg; Pkg.add("Random")` to install the Random package.
251
```

Doesn't running `Pkg.test("Phylo")` locally help to check this without waiting for CI?

**Comment by mkborregaard** [comment]
Is this the current state of affairs? Essentially Phylo is in an unstable state atm? Can we just get rid of JuliaDB?

**Comment by richardreeve** [comment]
Let's see what @claireh93's reworking of the tests achieves. We were using JuliaDB, which is why it's tested with that. When I run the tests locally on 1.6-beta1, there are no problems.

**Comment by mkborregaard** [comment]
yeah here neither I also ran the tests locally

**Comment by mkborregaard** [comment]
@richardreeve @tpoisot can this be merged now? Did @claireh93 rework the tests? I'd love to use this at a workshop on Monday (in fact I'd planned to) but I'm running into issues because Phylo isn't compatible with current DataFrames

**Comment by richardreeve** [comment]
Working through it now. We have a bit too much going on in the github actions now, but I'll trim it down...

**Comment by richardreeve** [comment]
@mkborregaard What is actually problematic about DataFrames? There's no changes to the code in this PR, but it works on my laptop without any constraints on DataFrames...

**Comment by mkborregaard** [comment]
Just the bounds I think. Because compathelper isn't activated before this PR it lags behind the current version

**Comment by richardreeve** [comment]
Okay, @mkborregaard - all of the tests pass now, and I've added Windows here too and removed appveyor. Do you want to see whether it works for your purposes or if I've missed something and we need to add in another test? DataFrames is not used in anger in the code - it's just the default storage type for leaves so there's no reason for it to matter. It was all just fixing things for the registrator (which may happen again I guess?).

**Pull request merged**